src/openeo_grass_gis_driver/actinia_processing/base.py

```diff
@@ -271,6 +276,141 @@ def process_node_to_actinia_process_chain(node: Node) -> Tuple[list, list]:
+    pc["module"] = node.process_id
+    pc["inputs"] = []
+    pc["flags"] = None
+
+    # input parameters
+    data_object = None
+    for key in node.arguments.keys():
+        # is key a valid actinia option ?
+        ao = None
+        # not very elegant
+        for item in module["parameters"]:
+            if item["name"] == key:
+                ao = item
+        if ao is None:
+            # warning, error, exception?
+            continue
+        # check if it is an input object
+        if isinstance(node.arguments[key], dict) and \
+           "from_node" in node.arguments[key]:
+            # input option comes from another node in the process graph
+	    # which output object in the set of output_objects?
+            value = list(node.get_parent_by_name(parent_name=key).output_objects)[0]
+            data_object = value
+        elif item["schema"]["type"] == "boolean":
+            # flag
+            if node.arguments[key] is True:
+                if pc["flags"] is None:
+                    pc["flags"] = key
+                else:
+                    pc["flags"] = pc["flags"] + key
+        else:
+            # option answer, treat as string
+            value = node.arguments[key]
+            param = {"param": key,
+                     "value": value}
+            pc["inputs"].append(param)
+
+    if pc["flags"] is None:
+        del pc["flags"]
+
+    # TODO: support modules that do not have input maps
+    if data_object is None:
+        raise Exception("No input data object for actinia process '%s'" % module_name)
+
+    # output parameters
+    if "returns" in module:
+        for key in node.arguments.keys():
+            # a suggested method to allow openeo users to select certain
+            # outputs if a module can generate several outputs:
+            # put a dummy entry in "arguments"
+
+            # find actinia option in "returns" of the
+            # actinia module description
+            ao = None
+            # not very elegant
+            for item in module["returns"]:
+                if item["name"] == key:
+                    ao = item
+            if ao is None:
+                continue
+            datatype = None
+            if ao["schema"]["subtype"] == "cell":
+                datatype = GrassDataType.RASTER
+            elif ao["schema"]["subtype"] == "vector":
+                datatype = GrassDataType.VECTOR
+            elif ao["schema"]["subtype"] == "strds":
+                datatype = GrassDataType.STRDS
+
+            if datatype is not None:
+                # note that key is added to the output name
+                # in order to distinguish between different outputs
+                # of the same module
+                output_object = DataObject(name=f"{data_object.name}_{process_name}_{key}", datatype=datatype)
+                param = {"param": key,
+                         "value": output_object.grass_name()}
+                pc["inputs"].append(param)
```

Comment:
you could use pc["outputs"] for that.

Reply:
`pc["outputs"]` is not used in any of the openeo process translations. Worth a separate ticket?

Reply:
Wouldn't change function, only semantic. I resolve this thread.